src/Tribe/Rewrite.php

```diff
@@ -248,18 +249,41 @@ public function get_bases( $method = 'regex' ) {
 		}
 
 		$tec = Tribe__Events__Main::instance();
+		$user_locale = get_user_locale();
+		$locale = get_locale();
+
+		/**
+		 * Filters the text domains that should be loaded to get the correctly localized base slugs.
+		 *
+		 * @since 3.11.2
+		 *
+		 * @param array<string,string> $text_domains A map from text domain to the path to the file containing the
+		 *                                           translations.
+		 */
+		$domains = apply_filters( 'tribe_events_rewrite_i18n_domains', array(
+			'default'             => true, // Default doesn't need file path
+			'the-events-calendar' => $tec->plugin_dir . 'lang/',
+		) );
+
+		if ( $user_locale !== $locale ) {
+			/*
+			 * The bases should be generated using the site locale, not the user locale.
+			 * Switch to the site locale and force the plugins to load the correct translations.
+			 */
+			$this->translations_loader->load( $locale, $domains );
+		}
 
 		$default_bases = [
-			'month'    => [ 'month', $tec->monthSlug ],
-			'list'     => [ 'list', $tec->listSlug ],
-			'today'    => [ 'today', $tec->todaySlug ],
-			'day'      => [ 'day', $tec->daySlug ],
-			'tag'      => [ 'tag', $tec->tag_slug ],
-			'tax'      => [ 'category', $tec->category_slug ],
+			'month'    => [ 'month', sanitize_title( __( 'month', 'the-events-calendar' ) ) ],
+			'list'     => [ 'list', sanitize_title( __( 'list', 'the-events-calendar' ) ) ],
+			'today'    => [ 'today', sanitize_title( __( 'today', 'the-events-calendar' ) ) ],
+			'day'      => [ 'day', sanitize_title( __( 'day', 'the-events-calendar' ) ) ],
```

Comment:
Can we make sure to memoize the values for `sanitize_title()` for all of these? Since this method can be called a ton of times due to having to figure out multiple links on a page. Same for the other PRs.